docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+- activity processing time (amount of time it takes to perform a particular service function - such as respond to a command request)
+
+### Definitions
+
+**Metric (or telemetry) data** is defined as the count or rate of some action, resource, or circumstance in the EdgeX instance or specific service.  Examples of metrics include:
+
+- the number of EdgeX Events sent from core data to an application service via message bus (or via device service to application service in Ireland and beyond)
+- the number of requests on a service API
+- the average time it takes to process a message through an application service
+- The number of errors logged by a service
+
+The collection and dissemination of metric data will require internal service level instrumentation (relevant to that service) to capture and send data about relevant EdgeX operations.  EdgeX does not currently offer any service instrumentation.
+
+### Metric Use
+
+As a first step in implementation of metrics data, EdgeX will make metric data available to other subscribing 3rd party applications and systems, but will not necessarily consume or use this information itself.  
+
+In the future, EdgeX may consume its own metric data.  For example, EdgeX may, in the future, use a metric on the number of EdgeX events being sent to core data as the means to throttle back device data collection.
+
+In the future, EdgeX application services may optionally subscribe to a service's metrics messages bus (by attaching to the appropriate message pipe for that service).  Thus allowing additional filtering, transformation, endpoint control of metric data from that service.  At the point where this feature is supported, consideration would need to be made as to whether all events (sensor reading messages and metric messages) go through the same application services.
+
+At this time, EdgeX will not persist the metric data (except as it may be retained as part of and message bus subsystem such as in an MQTT broker).  Consumers of metric data are responsible for persisting the data if needed, but this is external to EdgeX.  Persistence of metric information may be considered in the future based on requirements and adopter demand for such a feature.
+
+In general, EdgeX metrics are meant to provide external applications and systems better information about what is happening "inside" EdgeX services and the associated devices with which it communicates.
+
+### Requirements
+
+- Services will push specified metrics collected for that service to a specified (by configuration) message endpoint (as supported by the EdgeX  message bus implementation; currently either Redis Pub/Sub or MQTT implementations are supported)
+    - Each service will have configuration that specifies a message endpoint for the service metrics.  The metrics message topic communications may be secured or unsecured (just as application services provide the means to export to secured or unsecured message pipes today).

```

Comment:
I think this approach has some of the same issues as the recent North/South ADR. Namely, our default internal message bus implementation is secure by default. While it's possible today to access the message bus via an application service, it's not possible to (easily) grant permission to external clients. Doing this properly requires Vault support.

So does this mean the only way to access metrics from an external system would be via an application service which would act as a proxy/bridge to an external message bus, or do we add support to every service to allow an external message bus?